### PR TITLE
LameClient ready for absl::Status

### PR DESCRIPTION
**Pull request by veblush**
Instead of storing the pointer of `grpc_error` (which is not feasible for `absl::Status`) in the channel-arg, `grpc_error_handle*` is to be stored to support `absl::Status` case.

### CODE REVIEW

File: src/core/ext/filters/client_channel/dynamic_filters.cc [hunk 1]
```diff
@@ -163,10 +163,16 @@ RefCountedPtr<DynamicFilters> DynamicFilters::Create(
     // Channel stack creation failed with requested filters.
     // Create with lame filter instead.
     grpc_error_handle error = p.second;
-    grpc_arg error_arg = MakeLameClientErrorArg(error);
+    grpc_error_handle* error_ptr = nullptr;
+    if (error != GRPC_ERROR_NONE) {
+      error_ptr = new grpc_error_handle();
```
Comment (markdroth) [inline src/core/ext/filters/client_channel/dynamic_filters.cc:168]: I don't think we need any dynamic allocation here.  I think we can just do:

```
grpc_arg error_arg = MakeLameClientErrorArg(&error);
```

File: src/core/lib/surface/lame_client.cc [hunk 2]
```diff
@@ -176,10 +184,16 @@ grpc_channel* grpc_lame_client_channel_create(const char* target,
           GRPC_ERROR_INT_GRPC_STATUS, error_code),
       GRPC_ERROR_STR_GRPC_MESSAGE,
       grpc_slice_from_static_string(error_message));
-  grpc_arg error_arg = grpc_core::MakeLameClientErrorArg(error);
+  grpc_error_handle* error_ptr = nullptr;
+  if (error != GRPC_ERROR_NONE) {
+    error_ptr = new grpc_error_handle();
```
Comment (markdroth) [inline src/core/lib/surface/lame_client.cc:189]: Same thing here:

```
grpc_arg error_arg = grpc_core::MakeLameClientErrorArg(&error);
```

File: src/core/ext/filters/client_channel/dynamic_filters.cc (hunk 1 above)
Comment (veblush) [reply src/core/ext/filters/client_channel/dynamic_filters.cc]: Good catch! Done.

File: src/core/lib/surface/lame_client.cc (hunk 2 above)
Comment (veblush) [reply src/core/lib/surface/lame_client.cc]: Done.

**Review by markdroth — APPROVED** [review]
Looks great!

**Pull request merged**